Responses API errors out with using previous_response_id and tool use

### System Info

.

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### 🐛 Describe the bug

Repro:
```

# LS responses
response = client.responses.create(
    model=model_id,
    input=[
        {"role": "user", "content": "what's the weather in tokyo?"}
    ],
    tools=[
        {
            "type": "function",
            "name": "get_weather",
            "description": "Get the weather in a given city",
            "parameters": {
                "type": "object",
                "properties": {
                    "city": {"type": "string", "description": "The city to get the weather for"},
                },
            },
        }
    ],
    stream=False,
)
pprint(response)

response2 = client.responses.create(
    model=model_id,
    input=[
        {
        "type": "function_call_output",
        "call_id": response.output[0].call_id,
        "output": "sunny and warm"
        }
    ],
    previous_response_id=response.id,
    stream=False,
)
pprint(response2)
```

### Error logs

INFO:     ::1:53612 - "POST /v1/openai/v1/responses HTTP/1.1" 
  400 Bad Request
  19:05:00.492 [END] /v1/openai/v1/responses [StatusCode.OK] 
  (338.31ms)
   19:05:00.491 [ERROR] Error executing endpoint 
  route='/v1/openai/v1/responses' method='post'
  Traceback (most recent call last):
    File "/Users/erichuang/projects/llama-stack/llama_stack/dist
  ribution/server/server.py", line 239, in route_handler
      result = await maybe_await(value)
               ^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/erichuang/projects/llama-stack/llama_stack/dist
  ribution/server/server.py", line 174, in maybe_await
      return await value
             ^^^^^^^^^^^
    File "/Users/erichuang/projects/llama-stack/llama_stack/prov
  iders/inline/agents/meta_reference/agents.py", line 337, in 
  create_openai_response
      return await 
  self.openai_responses_impl.create_openai_response(
             
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/erichuang/projects/llama-stack/llama_stack/prov
  iders/inline/agents/meta_reference/openai_responses.py", line 
  356, in create_openai_response
      async for stream_chunk in stream_gen:
    File "/Users/erichuang/projects/llama-stack/llama_stack/prov
  iders/inline/agents/meta_reference/openai_responses.py", line 
  426, in _create_streaming_response
      completion_result = await 
  self.inference_api.openai_chat_completion(
                          
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/erichuang/projects/llama-stack/llama_stack/prov
  iders/utils/telemetry/trace_protocol.py", line 103, in 
  async_wrapper
      result = await method(self, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/erichuang/projects/llama-stack/llama_stack/dist
  ribution/routers/inference.py", line 545, in 
  openai_chat_completion
      response_stream = await 
  provider.openai_chat_completion(**params)
                        
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/erichuang/projects/llama-stack/llama_stack/prov
  iders/utils/telemetry/trace_protocol.py", line 103, in 
  async_wrapper
      result = await method(self, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/erichuang/projects/llama-stack/llama_stack/prov
  iders/remote/inference/fireworks/fireworks.py", line 384, in 
  openai_chat_completion
      return await 
  OpenAIChatCompletionToLlamaStackMixin.openai_chat_completion(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ^^^^^^^^^^^^^^^^
    File "/Users/erichuang/projects/llama-stack/llama_stack/prov
  iders/utils/inference/openai_compat.py", line 1368, in 
  openai_chat_completion
      messages = openai_messages_to_messages(messages)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/erichuang/projects/llama-stack/llama_stack/prov
  iders/utils/inference/openai_compat.py", line 1013, in 
  openai_messages_to_messages
      content=openai_content_to_content(message.content),
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/erichuang/projects/llama-stack/llama_stack/prov
  iders/utils/inference/openai_compat.py", line 1042, in 
  openai_content_to_content
      raise ValueError(f"Unknown content type: {content}")
  ValueError: Unknown content type: None

### Expected behavior

generates model response without error

--- a/llama_stack/providers/utils/inference/openai_compat.py
+++ b/llama_stack/providers/utils/inference/openai_compat.py
@@ -1026,7 +1026,9 @@ def openai_messages_to_messages(
     return converted_messages
 
 
-def openai_content_to_content(content: str | Iterable[OpenAIChatCompletionContentPartParam]):
+def openai_content_to_content(content: str | Iterable[OpenAIChatCompletionContentPartParam] | None):
+    if content is None:
+        return ""
     if isinstance(content, str):
         return content
     elif isinstance(content, list):

--- a/tests/integration/agents/test_openai_responses.py
+++ b/tests/integration/agents/test_openai_responses.py
@@ -221,3 +221,56 @@ def test_list_response_input_items_with_limit_and_order(openai_client, client_wi
             assert hasattr(item, "type")
             assert item.type == "message"
             assert item.role in ["user", "assistant"]
+
+
+@pytest.mark.skip(reason="Tool calling is not reliable.")
+def test_function_call_output_response(openai_client, client_with_models, text_model_id):
+    """Test handling of function call outputs in responses."""
+    if isinstance(client_with_models, LlamaStackAsLibraryClient):
+        pytest.skip("OpenAI responses are not supported when testing with library client yet.")
+
+    client = openai_client
+
+    # First create a response that triggers a function call
+    response = client.responses.create(
+        model=text_model_id,
+        input=[
+            {
+                "role": "user",
+                "content": "what's the weather in tokyo? You MUST call the `get_weather` function to find out.",
+            }
+        ],
+        tools=[
+            {
+                "type": "function",
+                "name": "get_weather",
+                "description": "Get the weather in a given city",
+                "parameters": {
+                    "type": "object",
+                    "properties": {
+                        "city": {"type": "string", "description": "The city to get the weather for"},
+                    },
+                },
+            }
+        ],
+        stream=False,
+    )
+
+    # Verify we got a function call
+    assert response.output[0].type == "function_call"
+    call_id = response.output[0].call_id
+
+    # Now send the function call output as a follow-up
+    response2 = client.responses.create(
+        model=text_model_id,
+        input=[{"type": "function_call_output", "call_id": call_id, "output": "sunny and warm"}],
+        previous_response_id=response.id,
+        stream=False,
+    )
+
+    # Verify the second response processed successfully
+    assert response2.id is not None
+    assert response2.output[0].type == "message"
+    assert (
+        "sunny" in response2.output[0].content[0].text.lower() or "warm" in response2.output[0].content[0].text.lower()
+    )

